test/without_coverage.jl

```diff
@@ -0,0 +1,27 @@
+using PrecompileTools
+using Test
+using InteractiveUtils
+
+@testset "No-coverage tests" begin
+    @assert Base.JLOptions().code_coverage == 0 "These tests should be run with coverage disabled"
+
+    push!(LOAD_PATH, @__DIR__)
+
+    using MSort
+    using AliasTables
+    x = rand(64)
+    pipe = Pipe()
+    oldstderr = stderr
+    redirect_stderr(pipe)
+    @trace_compile begin
```

Comment:
You have to be a bit careful here with the exact same thing this PR attempts to fix, that Julia doesn't compile the code in the `begin` block before the tracing has been enabled (as discussed in https://github.com/JuliaLang/julia/issues/57899). So maybe there should be an `@eval` here or something like that?

Reply:
It might work because (AFAIU), `@testset` has special code to insert world age bumps between statements, or something to that effect. Or because of some other completely different reason... 🤷